From ed7d2469613be7d94068ace83fcc1a986e353df4 Mon Sep 17 00:00:00 2001
From: Eric Naim <dnaim@cachyos.org>
Date: Tue, 10 Feb 2026 18:20:34 +0800
Subject: [PATCH 3/6] r8125: Add Kconfig and Makefile entries

Kconfig is based on r8169, IDK what the actual dependencies are.

Signed-off-by: Eric Naim <dnaim@cachyos.org>
---
 drivers/net/ethernet/realtek/Kconfig  | 15 +++++++++++++++
 drivers/net/ethernet/realtek/Makefile |  2 ++
 2 files changed, 17 insertions(+)

diff --git a/drivers/net/ethernet/realtek/Kconfig b/drivers/net/ethernet/realtek/Kconfig
index 272c83bfd..dc7cf96ad 100644
--- a/drivers/net/ethernet/realtek/Kconfig
+++ b/drivers/net/ethernet/realtek/Kconfig
@@ -95,6 +95,21 @@ config 8139_OLD_RX_RESET
 	  experience problems, you can enable this option to restore the
 	  old RX-reset behavior.  If unsure, say N.
 
+config R8125
+	tristate "Realtek 8125/8162 ethernet support"
+	depends on PCI
+	select FW_LOADER
+	select CRC32
+	select PHYLIB
+	select REALTEK_PHY
+	help
+	  Say Y here if you have a Realtek Ethernet adapter belonging to
+	  the following families:
+	  RTL8125 2.5GBit Ethernet
+
+	  To compile this driver as a module, choose M here: the module
+	  will be called r8125.  This is recommended.
+
 config R8169
 	tristate "Realtek 8169/8168/8101/8125 ethernet support"
 	depends on PCI
diff --git a/drivers/net/ethernet/realtek/Makefile b/drivers/net/ethernet/realtek/Makefile
index 046adf503..dee73dfd0 100644
--- a/drivers/net/ethernet/realtek/Makefile
+++ b/drivers/net/ethernet/realtek/Makefile
@@ -9,4 +9,6 @@ obj-$(CONFIG_ATP) += atp.o
 r8169-y += r8169_main.o r8169_firmware.o r8169_phy_config.o
 r8169-$(CONFIG_R8169_LEDS) += r8169_leds.o
 obj-$(CONFIG_R8169) += r8169.o
+r8125-y += r8125_n.o rtl_eeprom.o rtltool.o
+obj-$(CONFIG_R8125) += r8125.o
 obj-$(CONFIG_RTASE) += rtase/
-- 
2.53.0

